pkg/controller/installation/products/launcher/reconciler.go

```diff
@@ -0,0 +1,323 @@
+package launcher
+
+import (
+	"context"
+	"github.com/RHsyseng/operator-utils/pkg/olm"
+	launcherv1alpha2 "github.com/fabric8-launcher/launcher-operator/pkg/apis/launcher/v1alpha2"
+	aerogearv1 "github.com/integr8ly/integreatly-operator/pkg/apis/aerogear/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/pkg/controller/installation/marketplace"
+	"github.com/integr8ly/integreatly-operator/pkg/controller/installation/products/config"
+	"github.com/integr8ly/integreatly-operator/pkg/resources"
+	oauthv1 "github.com/openshift/api/oauth/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	appsv1Client "github.com/openshift/client-go/apps/clientset/versioned/typed/apps/v1"
+	"github.com/pkg/errors"
+	pkgerr "github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pkgclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	defaultInstallationNamespace        = "launcher"
+	defaultSubscriptionName             = "integreatly-launcher"
+	defaultLauncherName                 = "launcher"
+	launcherRouteName                   = "launcher"
+	clientId                            = "launcher"
+	clientSecret                        = clientId + "-client-secret"
+	oauthId                             = "launcher"
+	githubOauthAppCredentialsSecretName = "launcher-oauth-github"
+)
+
+type Reconciler struct {
+	*resources.Reconciler
+	installation  *v1alpha1.Installation
+	appsv1Client  appsv1Client.AppsV1Interface
+	Config        *config.Launcher
+	ConfigManager config.ConfigReadWriter
+	mpm           marketplace.MarketplaceInterface
+}
+
+func NewReconciler(configManager config.ConfigReadWriter, instance *v1alpha1.Installation, appsv1Client appsv1Client.AppsV1Interface, mpm marketplace.MarketplaceInterface) (*Reconciler, error) {
+	config, err := configManager.ReadLauncher()
+	if err != nil {
+		return nil, err
+	}
+
+	if config.GetNamespace() == "" {
+		config.SetNamespace(instance.Spec.NamespacePrefix + defaultInstallationNamespace)
+	}
+
+	return &Reconciler{
+		appsv1Client:  appsv1Client,
+		ConfigManager: configManager,
+		Config:        config,
+		mpm:           mpm,
+		installation:  instance,
+		Reconciler:    resources.NewReconciler(mpm),
+	}, nil
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, inst *v1alpha1.Installation, serverClient pkgclient.Client) (v1alpha1.StatusPhase, error) {
+	logrus.Infof("Reconciling %s", defaultLauncherName)
+
+	phase, err := r.ReconcileNamespace(ctx, r.Config.GetNamespace(), inst, serverClient)
+	if err != nil {
```

Comment:
Other products are using this convention:

https://github.com/integr8ly/integreatly-operator/blob/master/pkg/controller/installation/products/codeready/reconciler.go#L68-L71

It would be nice for them all to be similar

Reply:
:+1: